heterocl/ast/ir_builder.py

```diff
@@ -469,6 +439,7 @@ def build_func_op(self, op: ast.FuncOp, ip):
         # as the same argument object may be refered in multiple functions
         # we need to make sure that the result is not reused
         for arg in op.args:
+            arg.prev_result = arg.result
```

Comment:
what does `prev_result` do? Why do we save the previous build result of a func block argument?

Reply:
I didn't figure out a good way to fix it in this quick prototype. Since the results will be removed, when users call for example `hcl.partition(A)` on a `hcl.placeholder`, it will generate a new `alloc` op for `A`, while `A` has already been a func block argument.

Reply:
I see, let's do this in another PR